dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkApplicationListener.java

```diff
@@ -84,7 +85,7 @@ else if (event.getType() == RequestEvent.Type.RESP_FILTERS_START) {
                         commitTransaction();
                     } catch (Exception e) {
                         rollbackTransaction();
-                        this.<RuntimeException>rethrow(e);
+                        throw new MappableException(e);
```

Comment:
Well, if we don't wrap an exception
````java
try {
    commitTransaction();
} catch (RuntimeException e) {
    rollbackTransaction();
    throw e;
}
````
Then `testSqlExceptionIsHandled` fails with 500 error.

If we delibitraly throw `DataException` from a resource then the mapper works, because Jersey internally wraps the exception as `MappingException`. But, as I beleive, it does so only for resources, but not for application listeners.